src/app/services/storage.service.ts

```diff
@@ -0,0 +1,53 @@
+import { IResponseForecastWeather } from '../models/response';
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, Observable } from "rxjs";
+
+@Injectable({
+	providedIn: 'root',
+})
+export class StorageService {
```

Comment:
I guess you don't need to use State , just Storage like local Storage is enough but if you want you can